runtime/src/iree/builtins/device/BUILD

```diff
@@ -4,20 +4,100 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-load("//build_tools/bazel:build_defs.oss.bzl", "iree_runtime_cc_library")
+load("//build_tools/bazel:build_defs.oss.bzl", "iree_cmake_extra_content", "iree_runtime_cc_library")
+load("//build_tools/bazel:iree_bitcode_library.bzl", "iree_bitcode_library")
+load("//build_tools/embed_data:build_defs.bzl", "c_embed_data")
 
 package(
     default_visibility = ["//visibility:public"],
     features = ["layering_check"],
     licenses = ["notice"],  # Apache 2.0
 )
 
+#===------------------------------------------------------------------------===#
+# Common sources
+#===------------------------------------------------------------------------===#
+
+BITCODE_SRCS = [
+    "device_generic.c",
+]
+
+BITCODE_HDRS = [
+    "device.h",
+]
+
+#===------------------------------------------------------------------------===#
+# Application-integrated static library
+#===------------------------------------------------------------------------===#
+
 iree_runtime_cc_library(
     name = "device",
+    srcs = BITCODE_SRCS,
+    hdrs = BITCODE_HDRS,
+)
+
+#===------------------------------------------------------------------------===#
+# Compiler bitcode files
+#===------------------------------------------------------------------------===#
+
+iree_cmake_extra_content(
+    content = """
+if(NOT IREE_BUILD_COMPILER)
+  return()
+endif()
+""",
+    inline = True,
+)
```

Comment:
Should these targets live under `compiler/` instead of `runtime/`? As a general rule of thumb, I'd expect `runtime/` targets to be built using imported/host compiler tools and not just always skip when `IREE_BUILD_COMPILER` is `OFF`.

Reply:
blame bazel - it really doesn't want to touch these files unless it's in the same directory